### ISSUE TITLE
compare measures for reset_timestamps = FALSE with and without extra timestamp 0

### ISSUE THREAD

**Issue by sbrockhaus** [PascalKieslich/mousetrap#6]
This is just a small consistency check I did to compare the computed mouse-measures when the timestamps are not reset to zero. In the first case I just left the data as it is with the first timestamp != 0. In the second case, I added the data for timestamp 0 by hand by repeating the first observation. As you can see in the example code almost all measures are equal. The only measures with discrepancies are AD and vel_min_time.  
The "time at which minimum velocity occurred first" is always the first timestamp. For the AD, the differences are quite small. 

```javascript
library(mousetrap)
library(abind) ## function abind()

### read in data, set reset_timestamps = FALSE
mt_data <- mt_import_mousetrap(mt_example_raw, reset_timestamps = FALSE)
mt_data$trajectories[1:2, 1, 1:5]

mt_data <- mt_derivatives(mt_data)
mt_data <- mt_measures(mt_data)


## now add the timepoint 0 and set it to the first observation
time0 <- mt_data$trajectories[ , , 1, drop = FALSE]
time0[ , "timestamps", ] <- 0 

mt_data0 <- mt_data
mt_data0$trajectories <- abind(time0, mt_data0$trajectories)
mt_data0$trajectories[1:2, 1, 1:6]

mt_data0 <- mt_derivatives(mt_data0)
mt_data0 <- mt_measures(mt_data0)


###### compare the computed mouse-measures
all(mt_data$measures == mt_data0$measures)
which( colMeans(mt_data$measures == mt_data0$measures) != 1 )

## the measures AD and vel_min_time are different!

## AD: Average deviation from direct path
## only slightly different 
mt_data$measures$AD
mt_data0$measures$AD

## Time at which minimum velocity occurred first
## is always the first timestamp
mt_data$measures$vel_min_time
mt_data0$measures$vel_min_time
all(mt_data$trajectories[, 1, 1] == mt_data$measures$vel_min_time) 


###### compare the computed derivatives
mt_data$trajectories[1, , 1:6]
mt_data0$trajectories[1, , 1:6]
```

**Comment by PascalKieslich** [comment]
Thanks for looking into this!

So far, if the timestamps are not reset when importing the trajectories into mousetrap and then running mt_measures, the following assumption was made (as reported from mt_measures):

> Trajectories detected where first timestamp is greater than 0. Assuming period without movement starting at timestamp 0.

So, it was definitely a good idea to check explicitly whether all results are in line with this assumption. Thanks again for that!

Regarding the two discrepancies you found: I will talk about the AD discrepancy at the end. The vel_min_time discrepancy should not have occurred (and is due to the fact that when calculating time measures for velocity and acceleration, no additional timestamp with value 0 was added in case the first time stamp was greater than 0).

While looking at the checks you did and the discrepancy for vel_min_time, @FelixHenninger and I discussed about what should be the desired behavior of mt_measures in case that the first timestamp in a trial is greater than 0. In general, the timestamps are only relevant for the time related mouse-tracking measures (e.g., MAD_time, which corresponds to the time at which the maximum absolute deviation was reached first). So one should first think about what these measures actually indicate. To make this explicit, I added the following section to the function documentation:

> Note that all time related measures (except idle_time) are reported using the timestamp metric as present in the data. To interpret the timestamp values as time since tracking start, the assumption has to be made that for each trajectory the tracking started at timestamp 0 and that all timestamps indicate the time passed since tracking start. Therefore, all timestamps should be reset during data import by subtracting the value of the first timestamp from all timestamps within a trial (assuming that the first timestamp corresponds to the time when tracking started). Timestamps are reset by default when importing the data using one of the mt_import functions (e.g., mt_import_mousetrap).

After having made  this explicit, we think that mt_measures should not make an additional assumption (and corresponding adjustment of measures) for the special case that a trajectory contains a trial where the first timestamp is > 0.  Instead, it will always report the corresponding time measures based on the actual timestamps included. Accordingly, I adjusted the message reported by mt_measures for the special case to:

> Trajectories detected where first timestamp is greater than 0. Please see Details section of mt_measures documentation for interpretation of time related measures.

Relating to your example code this means that all time related measures that were 0 in the case where an additional first position with 0 was added, now still correspond to the value of the first  timestamp in the data, which leads to discrepancies for MD_below_time (in some cases), idle_time, and vel_min_time.

All of this is implemented in d61792757821857b30e84e5c7c4de6297608b3a4.

Regarding the average deviation: the AD calculation basically ignores all timestamp information (which led to the small discrepancies when you added another position). This is something I have also now made explicit in the function documentation (see 71c2879de9f1b2003f4304502b9f4e0d38f4b54d).
